test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -928,5 +941,62 @@ public async Task VerifyRestorePackageByBuilding()
             // Assert
             Directory.Exists(installedPackageFolderPath).Should().BeTrue();
         }
+
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public void VerifyPackageNotRestoredAfterDisablingPackageRestore()
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.IOSLibraryApp, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            _pathContext.Settings.DisableAutoRestore();
+            CommonUtility.RestoreNuGetPackages(VisualStudio, Logger);
+            VisualStudio.ObjectModel.Shell.ToolWindows.ErrorHub.ShowErrors();
+
+            // Assert
+            VisualStudio.AssertErrorListContainsSpecificError("NuGet restore is currently disabled.");
+        }
+
+        [DataTestMethod]
+        [Timeout(DefaultTimeout)]
+        public async Task VerifyPackageNotRestoredAfterDisablingAutomaticInPackageRestoreSection()
+        {
+            // Arrange
+            NuGetApexTestService nugetTestService = GetNuGetTestService();
+
+            await CommonUtility.CreatePackageInSourceAsync(_pathContext.PackageSource, TestPackageName, TestPackageVersionV1);
+
+            SolutionService solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution("TestSolution", _pathContext.SolutionRoot);
+            ProjectTestExtension project = solutionService.AddProject(ProjectLanguage.CSharp, ProjectTemplate.ConsoleApplication, "TestProject");
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, Logger);
+            NuGetUIProjectTestExtension uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(TestPackageName, TestPackageVersionV1);
+
+            // Act
+            _pathContext.Settings.DisableAutomaticInPackageRestoreSection();
+
+            var installedPackageFolderPath = Path.Combine(_pathContext.PackagesV2, "Contoso.A.1.0.0");
+            Directory.Exists(installedPackageFolderPath).Should().BeTrue();
```

Comment:
Act ideally doesn't include asserts. This is an assert.

Reply:
This test has three act steps as below, so we added the "//Act" to the first act site.  We are not sure if it is allowed to add more than one Act in the same case.

- Disable automatic in package restore section
- Delete the package in package folder
- Build the project

Reply:
You can have a preconditions part. 

The idea is to have 1 act.

Reply:
Ok, we have updated the "//Act" to "build the project" site.